hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/oauth/OAuthHandler.java

```diff
@@ -0,0 +1,14 @@
+package pl.allegro.tech.hermes.consumers.consumer.oauth;
+
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.consumers.consumer.sender.MessageSendingResult;
+
+public interface OAuthHandler {
```

Comment:
Right, I was planning to do some renaming/moving of the classes but didn't get down to it.

Now I've made some wider refactoring introducing `MessageDeliveryHandler` and `MessageDeliveryHandlersController` interfaces that simply react to success/failures of message delivery. They are completely isolated from the OAuth context. 
We have a single implementation of the controller which is aware of the OAuth context and manages a number of handlers that are created per-subscription. Each handler has access to the OAuth tokens cache and decides whether it should be updated.
I like it much more that way. :) What do you think @adamdubiel?